api/types/constants.go

```diff
@@ -138,6 +138,16 @@ const (
 	// SubKindOpenSSHNode is a registered OpenSSH (agentless) node.
 	SubKindOpenSSHNode = "openssh"
 
+	// SubKindOpenSSHEphemeralKeyNode is a registered OpenSSH (agentless) node that doesn't require trust in Teleport CA.
+	// For each session an SSH Key is created and uploaded to the target host using a side-channel.
+	//
+	// For Amazon EC2 Instances, it uploads the key using:
+	// https://docs.aws.amazon.com/ec2-instance-connect/latest/APIReference/API_SendSSHPublicKey.html
+	// This Key is valid for 60 seconds.
+	//
+	// It uses the private key created above to SSH into the host.
+	SubKindOpenSSHEphemeralKeyNode = "openssh-ephemeral-key"
```

Comment:
wonder if we should just make a subkind for ec2 instance connect explicitly. If we have another type of `SubKindOpenSSHEphemeralKeyNode` in the future, what fields/attributes we will use to differentiate between those?

Reply:
I can rename it to OpenSSHEC2InstanceConnectEndpoint (`openssh-ec2-ice`)
What do you think?

Reply:
> I can rename it to OpenSSHEC2InstanceConnectEndpoint (openssh-ec2-ice)

👍 

@codingllama what's your thought on an explicit subkind like this vs a slightly more generic `SubKindOpenSSHEphemeralKeyNode`?

Reply:
That's alright by me, you are the experts as far I'm concerned. (It does seem like a good idea.)